galleries/examples/text_labels_and_annotations/fonts_demo.py

```diff
@@ -21,43 +21,36 @@
 fig.text(0.1, 0.9, 'family', fontproperties=heading_font, **alignment)
 families = ['serif', 'sans-serif', 'cursive', 'fantasy', 'monospace']
 for k, family in enumerate(families):
-    font = FontProperties()
-    font.set_family(family)
+    font = FontProperties(family=family)
```

Comment:
This broke with `sans-serif`, probably because of this special case:
https://github.com/QuLogic/matplotlib/blob/0cdb45da1bfe0af330f87eeadcf633919cd26f80/lib/matplotlib/font_manager.py#L614-L620

Reply:
Whoops. That's surprising and awkward. The `__init__` parameter behaves differently compared to the `set_family` method. 😢 But it's documented.

I believe `FontProperties(pattern)` should be deprecated and we should instead add a `FontProperties.from_pattern(pattern)` method. But that's for another PR. For now, we have to go with the list.